src/serde_impl.rs

```diff
@@ -0,0 +1,42 @@
+use std::fmt;
+use std::str::FromStr;
+
+use serde::de::{self, Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+
+use super::Mime;
+
+impl Serialize for Mime {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.as_ref())
+    }
+}
+
+impl<'de> Deserialize<'de> for Mime {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct Visitor;
+
+        impl<'de> de::Visitor<'de> for Visitor {
+            type Value = Mime;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid MIME type")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Mime, E>
+            where
+                E: de::Error,
+            {
+                Mime::from_str(value).or_else(|e| Err(E::custom(format!("{}", e))))
```

Comment:
Cannot this be simply `Mime::from_str(value).or_else(|e| Err(E::custom(e)))`, as in, with removed `format!` call?